### PR TITLE
feat: refactor inference

### PR DESCRIPTION
**Pull request by stephantul**
This PR refactors inference and removes all external dependencies.

Specifically: we no longer use scikit-learn for evaluation metrics or inference. We also don't use skops for persistence any more. Skops specifically was no longer really maintained, despite having vulnerabilities. Instead, we save the models 

Roundtrip accuracy is maintained. Models are now saved using `safetensors`, with the `config.json` containing relevant information.

### CODE REVIEW

File: model2vec/train/utils.py [hunk 1]
```diff
@@ -81,15 +79,46 @@ def train_test_split(
...
         label_counts = Counter(y)
         if min(label_counts.values()) < 2:
             logger.info("Some classes have fewer than 2 samples. Stratification is disabled.")
-            stratify_data = None
-        else:
-            stratify_data = y
-    return sklearn_split(X, y, test_size=test_size, random_state=42, shuffle=True, stratify=stratify_data)  # type: ignore
+            stratify = False
+
+    train_indices: list[int]
```
Comment (Pringled) [inline model2vec/train/utils.py:92]: I think there is a bug in the new train/test split logic but I don't fully understand what it is (I ran the code because I didn't fully understand how it works); e.g. if I do this I get a different length for the test set:

```python
from sklearn.model_selection import train_test_split as sklearn_split
from model2vec.train.utils import train_test_split

X = [str(i) for i in range(100)]
y = [label for label in range(10) for _ in range(10)]

_, old_test, _, _ = sklearn_split(
    X, y, test_size=0.15, random_state=42, shuffle=True, stratify=y
)
_, new_test, _, _ = train_test_split(X, y, test_size=0.15)

print(len(old_test), len(new_test))  # 15, 20
```

**Review by Pringled — APPROVED** [review]
Nice, LGTM! Found one bug (I think), but overal looks much much nicer. I guess we should do a major version bump for this
